cpu/sam0_common/periph/rtc.c

```diff
@@ -65,71 +65,33 @@ static inline void _rtc_set_enabled(bool on)
 #ifdef CPU_SAMD21
 static void _rtc_clock_setup(void)
 {
-    /* RTC uses External 32,768KHz Oscillator (OSC32K isn't accurate enough p1075/1138)*/
-    SYSCTRL->XOSC32K.reg =  SYSCTRL_XOSC32K_ONDEMAND |
-                            SYSCTRL_XOSC32K_EN32K |
-                            SYSCTRL_XOSC32K_XTALEN |
-                            SYSCTRL_XOSC32K_STARTUP(6) |
-                            SYSCTRL_XOSC32K_ENABLE;
-
     /* Setup clock GCLK2 with OSC32K divided by 32 */
-    GCLK->GENDIV.reg = GCLK_GENDIV_ID(2)|GCLK_GENDIV_DIV(4);
-    GCLK->GENCTRL.reg = (GCLK_GENCTRL_GENEN | GCLK_GENCTRL_SRC_XOSC32K | GCLK_GENCTRL_ID(2) | GCLK_GENCTRL_DIVSEL );
-    GCLK->CLKCTRL.reg = (uint32_t)((GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK2 | (RTC_GCLK_ID << GCLK_CLKCTRL_ID_Pos)));
+    GCLK->GENDIV.reg = GCLK_GENDIV_ID(2) | GCLK_GENDIV_DIV(4);
+    GCLK->GENCTRL.bit.DIVSEL = 1;
+    GCLK->CLKCTRL.reg = GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN(2) | GCLK_CLKCTRL_ID_RTC;
     while (GCLK->STATUS.bit.SYNCBUSY) {}
 }
 #else
 static void _rtc_clock_setup(void)
 {
-    MCLK->APBAMASK.reg |= MCLK_APBAMASK_OSC32KCTRL;
-
-#if EXTERNAL_OSC32_SOURCE
-
-    /* RTC uses External 32,768KHz Oscillator */
-    OSC32KCTRL->XOSC32K.reg = OSC32KCTRL_XOSC32K_XTALEN
-                            | OSC32KCTRL_XOSC32K_EN1K
-                            | OSC32KCTRL_XOSC32K_RUNSTDBY
-                            | OSC32KCTRL_XOSC32K_ENABLE;
-
-    /* Wait XOSC32K Ready */
-    while (OSC32KCTRL->STATUS.bit.XOSC32KRDY==0) {}
-
     /* RTC source clock is external oscillator at 1kHz */
+#if EXTERNAL_OSC32_SOURCE
```

Comment:
please add #include "periph_conf." at the beginning of the file

Reply:
Ok.